src/base/bittorrent/session.cpp

```diff
@@ -4595,10 +4597,10 @@ void Session::handleSessionStatsAlert(const lt::session_stats_alert *p)
 
     emit statsUpdated();
 
-    if (m_refreshEnqueued)
-        m_refreshEnqueued = false;
-    else
-        enqueueRefresh();
+    QTimer::singleShot(1000, this, [this]()
```

Comment:
I would like to make refresh intervals for session stats 2s and remove 1-minute graph.

@Chocobo1 @glassez ?

Reply:
@jagannatharjun Why not make the session stats interval configurable, or perhaps tied to the transfer list refresh interval?

Reply:
that will require a lot of work and is not worth it.

Reply:
I don't think we should provide this statistics in realtime, so I don't mind increasing this interval.
Is it used somewhere other than graphs, where it might be sensitive to such a change?

Reply:
it's used in 2 more places, for mainwidow's tooltip and title text, and session statistics window, increasing interval should not be an issue I guess, but our speed plotting mechanism actually assumes a 1-second refresh interval so that requires some work which I already have a patch for. also Deluge also uses 2s interval for stats update

Reply:
here's a question out of ignorance; is there anything preventing the `Session` object from being destructed while this timer is active? resulting in the callback being called and `m_nativeSession` being a dangling pointer.

Reply:
> here's a question out of ignorance; is there anything preventing the Session object from being destructed while this timer is active? resulting in the callback being called and m_nativeSession being a dangling pointer.

the function is called in the main loop, so afaik this won't be called after the Session object is destructed (the timer is tied to the lifetime of session object) or while it's being destructed since only one function can execute at a time.

Reply:
> I would like to make refresh intervals for session stats 2s and remove 1-minute graph.

I've been using this PR in my main setup for more than 2 months now and felt that 1-minute graph is actually useful plus graphs with 2s update intervals don't look good so in the latest revision, I changed the update interval back to 1s.